certif/tests/acceptance/session-details-certification-candidates_test.js

```diff
@@ -315,6 +315,18 @@ module('Acceptance | Session Details Certification Candidates', function (hooks)
             assert.dom('[data-test-notification-message="success"]').hasText('Le candidat a été ajouté avec succès.');
           });
 
+          test('it should disable the button during API call', async function (assert) {
+            // when
+            const screen = await visitScreen(`/sessions/${session.id}/candidats`);
+            await click(screen.getByRole('button', { name: 'Ajouter un candidat' }));
+            await _fillFormWithCorrectData(screen);
+            const addButton = screen.getByRole('button', { name: 'Ajouter le candidat' });
+            await click(addButton);
+
+            // then
+            assert.ok(addButton.getAttribute('aria-disabled'));
```

Comment:
question : possible que ce soit flaky ça ou pas ? Si l'appel mirage est résolu plus vite que le assert ? 🤷🏿

Reply:
Carrément. J'ai posté sur Slack pour savoir comment tester proprement, pas eu de réponse.
Je propose de merger et de virer si c'est flaky !

Reply:
Ca devrait fonctionner:
```javascript
const addButton = screen.getByRole('button', { name: 'Ajouter le candidat' });
server.post('/sessions/:id/certification-candidates', function (schema, request) {
      // on fait l'assert pendant la requete, s'assurer que le notOk ne passerait pas ici
      assert.ok(addButton.getAttribute('aria-disabled'));
  });
await click(addButton);
assert.notOk(addButton.getAttribute('aria-disabled'));
```

Reply:
J'ai essayé dans un commit séparé, j'obtiens l'erreur 
"Promise rejected after "it should disable the button during API call": Assertion Failed: You can only unload a record which is not inFlight. `<certification-candidate:null>`"

C'est peut être dû à la fin prématurée du test, mais comment gérer le cas ?
https://github.com/emberjs/data/issues/3537

Reply:
Même problème avec 
```js
          test('it should disable the button during API call and enable it again afterwards', async function (assert) {
            // given
            let buttonHasBeenDisabled;
            assert.expect(2);
            const screen = await visitScreen(`/sessions/${session.id}/candidats`);
            await click(screen.getByRole('button', { name: 'Inscrire un candidat' }));
            await _fillFormWithCorrectData(screen);
            const addButton = screen.getByRole('button', { name: 'Inscrire le candidat' });
            server.post('/sessions/:id/certification-candidates', function () {
              buttonHasBeenDisabled = addButton.getAttribute('aria-disabled');
            });

            // when
            await click(addButton);

            // then
            assert.strictEqual(buttonHasBeenDisabled, 'false');
            assert.strictEqual(addButton.getAttribute('aria-disabled'), 'false');
          });
```
Output
```
<li class="pass"><span class="test-message">okay</span><span class="runtime">@ 979 ms</span></li><li class="pass"><span class="test-message">okay</span><span class="runtime">@ 979 ms</span></li><li class="fail"><span class="test-message">Attempted to handle event `deleteRecord` on &lt;certification-candidate:null&gt; while in state root.loaded.created.inFlight. </span><span class="runtime">@ 979 ms</span>

Source: | _unhandledEvent@http://localhost:4203/assets/vendor.js:117243:13 send@http://localhost:4203/assets/vendor.js:117048:14 deleteRecord@http://localhost:4203/assets/vendor.js:116515:12 deleteRecord@http://localhost:4203/assets/vendor.js:103595:27 _handleSavingError@http://localhost:4203/assets/pix-certif.js:2109:30 _handleUnknownSavingError@http://localhost:4203/assets/pix-certif.js:2098:12 saveCertificationCandidate@http://localhost:4203/assets/pix-certif.js:2042:16
-- | --


</li><li class="fail"><span class="test-message">Promise rejected after "it should disable the button during API call and enable it again afterwards": Assertion Failed: You can only unload a record which is not inFlight. `&lt;certification-candidate:null&gt;`</span></li>
```

Je garde la première version, si c'est flaky on revert

Reply:
Bon, la version d'origine ne passe plus :exploding_head: 
Je supprime le test et on merge tel quel, WDYT ?

Reply:
Allez, je merge